test/unit/org/apache/cassandra/metrics/TrieMemtableMetricsTest.java

```diff
@@ -0,0 +1,215 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+import org.apache.cassandra.SchemaLoader;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.config.OverrideConfigurationLoader;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.dht.Murmur3Partitioner;
+import org.apache.cassandra.dht.Range;
+import org.apache.cassandra.dht.Splitter;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.service.EmbeddedCassandraService;
+import org.apache.cassandra.service.StorageService;
+import org.jboss.byteman.contrib.bmunit.BMRule;
+import org.jboss.byteman.contrib.bmunit.BMRules;
+import org.jboss.byteman.contrib.bmunit.BMUnitRunner;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(BMUnitRunner.class)
+public class TrieMemtableMetricsTest extends SchemaLoader
+{
+    private static final int NUM_SHARDS = 13;
+
+    private static Logger logger = LoggerFactory.getLogger(TrieMemtableMetricsTest.class);
+    private static Session session;
+
+    private static final String KEYSPACE = "triememtable";
+    private static final String TABLE = "metricstest";
+
+    @BeforeClass
+    public static void loadSchema() throws ConfigurationException
+    {
+        // shadow superclass method; we'll call it directly
+        // after tinkering with the Config
+    }
+
+    @BeforeClass
+    public static void setup() throws ConfigurationException, IOException
+    {
+        OverrideConfigurationLoader.override((config) -> {
+            config.partitioner = "Murmur3Partitioner";
+        });
+        System.setProperty("cassandra.trie.memtable.shard.count", "" + NUM_SHARDS);
+
+        SchemaLoader.loadSchema();
+
+        Schema.instance.clear();
+
+        EmbeddedCassandraService cassandra = new EmbeddedCassandraService();
+        cassandra.start();
+
+        Cluster cluster = Cluster.builder().addContactPoint("127.0.0.1").withPort(DatabaseDescriptor.getNativeTransportPort()).build();
+        session = cluster.connect();
+
+        session.execute(String.format("CREATE KEYSPACE IF NOT EXISTS %s WITH replication = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 };", KEYSPACE));
+    }
+
+    private ColumnFamilyStore recreateTable()
+    {
+        return recreateTable(TABLE);
+    }
+
+    private ColumnFamilyStore recreateTable(String table)
+    {
+        session.execute(String.format("DROP TABLE IF EXISTS %s.%s", KEYSPACE, table));
+        session.execute(String.format("CREATE TABLE IF NOT EXISTS %s.%s (id int, val1 text, val2 text, PRIMARY KEY(id, val1)) WITH MEMTABLE = {'class':'TrieMemtable'};", KEYSPACE, table));
```

Comment:
It seems that there is a syntax error here:
```suggestion
        session.execute(String.format("CREATE TABLE IF NOT EXISTS %s.%s " +
                                      "(id int, val1 text, val2 text, PRIMARY KEY(id, val1)) " +
                                      "WITH MEMTABLE = 'trie'", 
                                      KEYSPACE, table));
```